res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -592,15 +591,41 @@ TraktorS2MK3.fxHandler = function (field) {
         return;
     }
 
+    const availableGroups = [ "[Channel1]", "[Channel2]" ];
+    var targetGroups = [];
     var fxNumber = parseInt(field.id[field.id.length - 1]);
-    var group = "[EffectRack1_EffectUnit" + fxNumber + "]";
 
-    // Toggle effect unit
-    TraktorS2MK3.fxButtonState[fxNumber] = !TraktorS2MK3.fxButtonState[fxNumber];
+    // Target decks whose shift button is pressed.
+    for (let group of availableGroups) {
+        if (TraktorS2MK3.shiftPressed[group]) targetGroups.push(group);
+    }
+
+    // If no shift button was pressed fall back to target all decks.
+    if (targetGroups.length === 0) targetGroups = availableGroups;
+
+    for (let group of targetGroups) {
+        engine.setValue(`[QuickEffectRack1_${group}]`, "loaded_chain_preset", fxNumber);
+    }
+};
+
+TraktorS2MK3.fxOutputHandler = function () {
+    const fxButtonCount = 4;
+    const availableGroups = [ "[Channel1]", "[Channel2]" ];
 
-    engine.setValue(group, "group_[Channel1]_enable", TraktorS2MK3.fxButtonState[fxNumber]);
-    engine.setValue(group, "group_[Channel2]_enable", TraktorS2MK3.fxButtonState[fxNumber]);
-    TraktorS2MK3.outputHandler(TraktorS2MK3.fxButtonState[fxNumber], field.group, "fxButton" + fxNumber);
+    var activeFx = {};
+    for (let group of availableGroups) {
+        activeFx[group] = engine.getValue(`[QuickEffectRack1_${group}]`, "loaded_chain_preset");
+    }
+
+    /* There is no way on the controller to indicate which deck the effect applies *
+     * to, but keeping both lit indicates that different effects are in use.       */
+    for (let fxButton = 1; fxButton <= fxButtonCount; ++fxButton) {
+        let active = false;
+        for (let group of availableGroups) {
+            active = active || activeFx[group] === fxButton;
```

Comment:
```suggestion
            active ||= (activeFx[group] === fxButton);
```
Maybe this would be better readable?